lib/lte_link_control/lte_lc.c

```diff
@@ -250,23 +300,23 @@ static void at_handler_cereg(const char *response)
 		return;
 	}
 
-	err = lte_lc_psm_get(&psm_cfg.tau, &psm_cfg.active_time);
-	if (err) {
-		if (err != -EBADMSG) {
-			LOG_ERR("Failed to get PSM information");
-		}
+	if (psm_cfg.tau == -1) {
+		/* Need to get legacy TAU using AT%XMONITOR.
+		 *
+		 * As we are in an AT notification handler that is run from the system work queue,
+		 * we shall not send AT commands here because another AT command might be ongoing,
+		 * and the second command will be stuck until the first one completes.
+		 * Sending any further AT notifications will gradually exhaust AT monitor
+		 * library's heap and eventually it will run out causing assert or fatal errors.
```

Comment:
What do you refer to with "fatal errors"? To the case when asserts are disabled? If asserts are disabled and the heap runs out, notifications will simply not be delivered, right? You could just say that in the comment, I think, for example:
```suggestion
		 * Further AT notifications from the modem will gradually exhaust AT monitor
		 * library's heap and eventually it will run out causing an assert or AT notifications
		 * not being dispatched.
```

Reply:
Yep, we should just say that AT notifications are not dispatched.